### PR TITLE
Change: merge Nginx update to 1.16.0

### PR DESCRIPTION
**Pull request by MengqiWu**
update Nginx core code to Nginx 1.16.0 from

https://github.com/nginx/nginx/compare/release-1.15.9...branches/stable-1.16

### CODE REVIEW

File: src/core/ngx_inet.c [hunk 1]
```diff
@@ -1695,44 +1630,68 @@ ngx_inet_resolve_host(ngx_pool_t *pool, ngx_url_t *u)
         }
 #endif
 
-    } else {
+#endif /* NGX_HAVE_GETADDRINFO && NGX_HAVE_INET6 */
```
Comment (chobits) [inline src/core/ngx_inet.c]: no return value for function ngx_inet_resolve_host

Comment (chobits) [reply src/core/ngx_inet.c]: diff your branch with nginx-1.16.0:

```

-#if (T_NGX_DNS_RESOLVE_BACKUP)
-        ph = (u_char *) getenv(NGX_DNS_RESOLVE_BACKUP_PATH);
-        if (ph != NULL) {
-            path.data = ph;
-            path.len = ngx_strlen(ph);
-            if (ngx_resolve_backup(pool, &u, path) != NGX_OK) {
-                ngx_log_error(NGX_LOG_WARN, ngx_cycle->log, 0,
-                              "dom %V backup local dns cache failed",
-                              &u->host);
-            }
-        }
-#endif
+    return NGX_OK;                                                <<< nginx-1.16.0
+}

 #endif /* NGX_HAVE_GETADDRINFO && NGX_HAVE_INET6 */

```

**Comment by chobits** [comment]
others diff with nginx-1.16.0 ACK

**Comment by chobits** [comment]
# Test Case for ngx_http_proxy_connect_module
* Tested-by: Xiaochen Wang
* Date: Mon Jun 17 10:45:07 CST 2019
* pwd: /Users/xiaochen/work/github/tengine-1.16.0
* cases: `modules/ngx_http_proxy_connect_module/t`
* git log:

```
commit 858ff3396993af84270305c3b7b5da3d8330f9b8 (HEAD -> master, origin/master, origin/HEAD)
Author: MengqiWu <mengqi.wmq@alibaba-inc.com>
Date:   Fri Jun 14 15:30:05 2019 +0800

    Change: merge Nginx update to 1.16.0
```

* case result:

```
$ TEST_NGINX_UNSAFE=yes TEST_NGINX_BINARY=/Users/xiaochen/work/github/tengine-1.16.0/objs/nginx prove -I /Users/xiaochen/work/github/tengine-1.16.0/../nginx-tests/lib modules/ngx_http_proxy_connect_module/t
modules/ngx_http_proxy_connect_module/t/http_proxy_connect.t .......... All 16 subtests passed
modules/ngx_http_proxy_connect_module/t/http_proxy_connect_timeout.t .. All 4 subtests passed
```

**Comment by chobits** [comment]
ACK

**Comment by wangfakang** [comment]
ACK.

**Pull request merged**